docs/best_practices.md

```diff
@@ -0,0 +1,153 @@
+
+> **Tip:** If you have multiple perspectives to share on a data set, consider providing multiple Dashboards rathering than trying to put everything into one. With Templates, there's no limit on how many you can provide.
+
+### Bucket Choices
+
+The easiest option is to provide a uniquely named [Bucket](https://v2.docs.influxdata.com/v2.0/reference/key-concepts/data-elements/#bucket) to be used by your Dashboard and (optional) Telegraf configurations. This simplifies things for the user, but at the expense of flexibility in reusing an existing Bucket. Another thing to consider is that the InfluxDB Cloud free tier has a limit of the number of Buckets a user can have, so adding a new one with your Template might not be an option for those users.
+
+A common option in InfluxDB is to use a Bucket named `telegraf` for all data coming in from the Telegraf agent. Existing users are likely to already have a Bucket by this name that your Template can use. But be sure to let them know in your Template's `README.md` that they will need to create a Bucket with this name if they don't already have one.
+
+#### Reusing existing Buckets
+The most flexible option is to let the user choose a Bucket instead. You can do this with a Variable that will provide them a drop-down list of their existing Buckets to choose from. Simply create a Query Variable called `bucket` with the following Flux query. Our documentation has some [common Variable queries](https://v2.docs.influxdata.com/v2.0/visualize-data/variables/common-variables/) you might find useful.
+
+```
+buckets()
+  |> filter(fn: (r) => r.name !~ /^_/)
+  |> rename(columns: {name: "_value"})
+  |> keep(columns: ["_value"])
+```
+> **Note:** the `filter()` call removes the system Buckets `_monitoring` and `_tasks` from the list
+
+Then, in your Dashboard you can read from the user's chosen Bucket in a cell query like this:
+
+```
+from(bucket: v.bucket) 
+```
+
+Once a Variable is used in a Dashboard, it will be shown to the user at the top of the page as a drop-down menu where they can change which value to use for that Variable.
+
+![Bucket Variable](img/bucket_variable.png)
+
+> **Tip:** If you are including a Telegraf configuration in your Template, be sure to use an environment variable such as `$INFLUX_BUCKET` so the user can define which Bucket they want it to send data to.
+
+### Using Variables
+
+[Variables](https://v2.docs.influxdata.com/v2.0/visualize-data/variables/) are a great way to give users flexibility over your Template without them having to make any changes to it. In the previous section you saw how a Variable can be used to display a list of Buckets to use, but they can also provide static or dynamic options anywhere you use Flux. You can create a Variable fromthe `Settings` -> `Variables` page of the InfluxDB UI.
```

Comment:
[**Dashboard** variables](https://v2.docs.influxdata.com/v2.0/visualize-data/variables/) ~are a great way to~ give users flexibility over your Template without ~them~ having to make any changes ~to it~. ~In the previous section you saw how a Variable can be used to display a list of Buckets to use, but they can also provide static or dynamic options anywhere you use Flux. You can~ Create a ~Variable~ **variable** ~fromthe~ **from the** `Settings` -> `Variables` page of the InfluxDB UI.